### PR TITLE
Crowdin: represent whitespace in code instead of in string

### PR DESCRIPTION
**Pull request by luzpaz**
Removing possible whitespace from strings lead to more accurate translations

### CODE REVIEW

**Comment by luzpaz** [comment]
@paullee0 do you mind taking a look at https://github.com/FreeCAD/FreeCAD/pull/4557/files#diff-3ec595fa09c804aea2062712c218dd2a145fb936b846a9671b67f244d96db9f4 ?

**Comment by paullee0** [comment]
> @paullee0 do you mind taking a look at https://github.com/FreeCAD/FreeCAD/pull/4557/files#diff-3ec595fa09c804aea2062712c218dd2a145fb936b846a9671b67f244d96db9f4 ?

OK, you mean the commit to delete a few whitespace in ArchStairs.py right?
Though not sure why there had been whitespace, and why it is not desirable thus removed, do not see any problem doing so :)
Anything you expect?

**Comment by luzpaz** [comment]
@yorikvanhavre when review is complete, will this go in to the 0.19 or 0.20 ?

**Comment by luzpaz** [comment]
@chennes can you review some of these. I know some of them are wrong.

File: src/Mod/Sketcher/Gui/ViewProviderSketch.cpp [hunk 1]
```diff
@@ -6493,29 +6493,33 @@ void ViewProviderSketch::UpdateSolverInformation()
         std::string msg;
         SketchObject::appendConflictMsg(getSketchObject()->getLastConflicting(), msg);
         signalSetUp(QString::fromLatin1("<font color='red'>%1<a href=\"#conflicting\"><span style=\" text-decoration: underline; color:#0000ff; background-color: #F8F8FF;\">%2</span></a><br/>%3</font><br/>")
-                    .arg(tr("Over-constrained sketch "))
+                    .arg(tr("Over-constrained sketch"))
+                    .arg(" ")
```
Comment (chennes) [inline src/Mod/Sketcher/Gui/ViewProviderSketch.cpp:6497]: This isn't right - you can't add more calls to .arg unless they correspond to added placeholders (the `%n` things) in the original string. You need to edit the HTML string above to include the spaces in it (that is, in this case, to add a space after the "%1").

Comment (luzpaz) [reply src/Mod/Sketcher/Gui/ViewProviderSketch.cpp]: Thanks, fixed in 0c99c23

**Comment by luzpaz** [comment]
@yorikvanhavre can you review this and after merge push to crowdin (since there are a lot of errors fixed + translations now available in the StartWB (see #4606 )

**Comment by luzpaz** [comment]
Travis is complaining about https://travis-ci.org/github/FreeCAD/FreeCAD/jobs/762361197#L7550-L7563  
https://github.com/FreeCAD/FreeCAD/commit/fedb8d3946e893bf5d3d5cf3731223908e58e8b4#diff-8883e954b4828659c14024871fb8a5af47ca0149506decf625c5e0fdc92491b2  
Any insight?

**Comment by luzpaz** [comment]
Ok, fixes pass CI. Please do final review

**Comment by luzpaz** [comment]
@wwmayer how would you prefer I squash this PR ?

**Comment by yorikvanhavre** [comment]
I would prefer to merge these things after 0.19 is fully out, in case we need to pull translations once more. If this PR is in (and the strings have been sent to crowdin), the translations we get from crowdin will differ from the 0.19 sources...

**Review by yorikvanhavre — APPROVED** [review]

**Comment by wwmayer** [comment]
@yorikvanhavre feel free to merge it when you think the time is right

File: src/Mod/Draft/importDXF.py [hunk 2]
```diff
@@ -3860,7 +3860,7 @@ def export(objectslist, filename, nospline=False, lwPoly=False):
                     filename = filename.encode("utf8")
             dxf.saveas(filename)
 
-        FCC.PrintMessage("successfully exported " + filename + "\n")
+        FCC.PrintMessage("successfully exported" + " " + filename + "\n")
```
Comment (yorikvanhavre) [inline src/Mod/Draft/importDXF.py:3863]: Better to add a translate() here too I guess

File: src/Mod/OpenSCAD/exportCSG.py [hunk 3]
```diff
@@ -269,4 +269,4 @@ def export(exportList,filename):
     csg.write("}\n}\n")
     # close file              
     csg.close()
-    FreeCAD.Console.PrintMessage("successfully exported "+filename)
+    FreeCAD.Console.PrintMessage("successfully exported" + " " + filename)
```
Comment (yorikvanhavre) [inline src/Mod/OpenSCAD/exportCSG.py:272]: first part should go into a translate()

File: src/Mod/Part/Gui/TaskCheckGeometry.cpp [hunk 4]
```diff
@@ -149,7 +149,7 @@ QString checkStatusToString(const int &index)
     }
     if (index > 33 || index < 0)
     {
-        QString message(QObject::tr("Out Of Enum Range: "));
+        QString message(QObject::tr("Out Of Enum Range:") + QString::fromLatin1(" "));
```
Comment (yorikvanhavre) [inline src/Mod/Part/Gui/TaskCheckGeometry.cpp:152]: IMHO if you touch not only string contents but also code itself, you should not skip CI

File: src/Mod/Start/Gui/Resources/translations/StartPage.ts [hunk 5]
```diff
@@ -343,11 +343,6 @@
         <source>An optional image to display as background</source>
         <translation type="unfinished"></translation>
     </message>
-    <message>
-        <location filename="../../DlgStartPreferences.ui" line="289"/>
-        <source>If this is checked, if a style sheet is specified in General preferences, it will be used and override the colors below</source>
-        <translation type="unfinished"></translation>
-    </message>
```
Comment (yorikvanhavre) [inline src/Mod/Start/Gui/Resources/translations/StartPage.ts:350]: Is that intentional?

File: src/Mod/Draft/importDXF.py (hunk 2 above)
Comment (luzpaz) [reply src/Mod/Draft/importDXF.py:3863]: @yorikvanhavre there are no other `translate()`'s in the `PrintMessage()` in the rest of this file so maybe that would be a separate PR?

File: src/Mod/OpenSCAD/exportCSG.py (hunk 3 above)
Comment (luzpaz) [reply src/Mod/OpenSCAD/exportCSG.py:272]: @yorikvanhavre there are no other translate()'s in the PrintMessage() in the rest of this file so maybe that would be a separate PR?

File: src/Mod/Part/Gui/TaskCheckGeometry.cpp (hunk 4 above)
Comment (luzpaz) [reply src/Mod/Part/Gui/TaskCheckGeometry.cpp:152]: fair enough

File: src/Mod/Start/Gui/Resources/translations/StartPage.ts (hunk 5 above)
Comment (luzpaz) [reply src/Mod/Start/Gui/Resources/translations/StartPage.ts:350]: Yes, this is a ghost translation.

**Comment by luzpaz** [comment]
If there are no other comments, this PR should be ready to go

File: src/Mod/Start/Gui/Resources/translations/StartPage.ts (hunk 5 above)
Comment (luzpaz) [reply src/Mod/Start/Gui/Resources/translations/StartPage.ts:350]: I mention it in the commit message:  
> For some reason updatets.py is not successfully removing obsolete translations. This is a long standing issue. The workaround is to manually remove them from the .ts file.
>  
> ref: https://forum.freecadweb.org/viewtopic.php?f=21&t=51825&p=483474#p482519  
> crowdin: https://crowdin.com/translate/freecad/7254/en-pl?filter=basic&value=0#6572886

File: src/Mod/OpenSCAD/exportCSG.py (hunk 3 above)
Comment (yorikvanhavre) [reply src/Mod/OpenSCAD/exportCSG.py:272]: sure!

**Pull request merged**